### PR TITLE
refactoring all explorer views to do server side rendering

### PR DESCRIPTION
**Pull request by vsoch**
This is a PR to refactor all current explorer views to use server side rendering (instead of a previous strategy to paginate), re-organize the API module to be more intuitive to find app-specific views (e.g., a server side rendering endpoint for conditions should be expected to be found in apps/api/conditions.py), and to do some other interface tweaks.

Signed-off-by: vsoch <vsochat@stanford.edu>

### CODE REVIEW

**Comment by vsoch** [comment]
Deployed to staging.

File: yeastphenome/apps/api/conditions.py [hunk 1]
```diff
@@ -75,6 +77,19 @@ def get(self, request):
...
             count = len(queryset)
 
+        # Create field to sort phenotypes (2) and conditions (3)
+        if queryset:
+            queryset = queryset.annotate(
+                phenotype_list=GroupConcat(
+                    "condition__conditionset__dataset__phenotype__name", ", "
+                )
+            ).distinct()
+            queryset = queryset.annotate(
```
Comment (abarysh) [inline yeastphenome/apps/api/conditions.py:91]: It turns out the solution is much simpler. We don't even need GroupConcat -- Django as a series of postgres-specific aggregates: https://docs.djangoproject.com/en/3.1/ref/contrib/postgres/aggregates/#stringagg

In our case all we need to do is:
```
from django.contrib.postgres.aggregates.general import StringAgg
agg_field = 'condition__conditionset__dataset__paper__first_author'
queryset = queryset.annotate(paper_list=StringAgg(agg_field, delimiter=', ', distinct=True, ordering=agg_field))
```

Comment (abarysh) [reply yeastphenome/apps/api/conditions.py:91]: For each condition, this will make a string of unique first authors in alphabetical order.

**Comment by vsoch** [comment]
That does the trick! https://github.com/yeastphenome/yeastphenome.org/pull/64/commits/a7247a3e500dd4f772021b293372b70fd4a05637

I also updated the ordering in the various functions for each models's papers() or phenotypes() - the result should be sorted as well otherwise the queries sorting won't match the function sorting. It looks like the Phenotypes conditions column is still slightly off, and I think it's because we filter down to condition types with `condition__conditionset__dataset__paper__latest_data_status__status__lt=10,` and this isn't done for the queryset produced during the explorer search. There could be something else, but I haven't tested. Should we add this filter to our explorer as well to try and make them the same?

```python
    def conditiontypes(self):
        # Unusual specification of "not relevant" papers because, in the other way,
        # conditiontypes were being filtered out if they were associated with a "not relevant" paper at least once
        return (
            apps.get_model("conditions", "ConditionType")
            .objects.filter(
                condition__conditionset__dataset__phenotype__observable=self,
                condition__conditionset__dataset__paper__latest_data_status__status__lt=10,
            )
            .order_by("name")
            .distinct()
```

And the changed just pushed are deployed to staging!

**Comment by abarysh** [comment]
Yeah, we should apply the same restriction on the condition types and not relevant papers. Just a couple of suggestions:

`status__lt=10` seems to be a bit error-prone (we want to exclude status #10, not include all statuses < 10 -- what about statuses > 10?). I would suggest to re-write the query as:

`.objects.filter(condition__conditionset__dataset__phenotype__observable=self).exclude(condition__conditionset__dataset__paper__latest_data_status__status__name='not relevant')`

(Added the name of the status to make it more obvious what is being filtered).

Also, should `.distinct()` come before `.order_by('name')`? It seems to me that, that way, there're fewer elements to sort, so in some cases it would work faster.
`

**Comment by vsoch** [comment]
Okay, the query is updated! The sorting is still off - they seem to both be sorted based on the ConditionType name, and they filter out the non relevant papers. The only difference is that the model query (for conditiontypes()) uses Observable as the base to start querying from, and the search view starts with ConditionType directly. I'm sort of stumped.

> Also, should .distinct() come before .order_by('name')? It seems to me that, that way, there're fewer elements to sort, so in some cases it would work faster.

Absolutely right - I think I have a habit of putting distinct at the end because it's typically the last thing, but definitely it's more efficient to sort a shorter list!

Changes above are live on staging.

**Comment by abarysh** [comment]
Is this still a relevant issue? If yes, could you show me the 2 queries that give you different sorting results?

**Comment by vsoch** [comment]
I just had a major rubber duck moment! I was writing up the issue carefully, and I realized that the solution wasn't to try and make the two queries the same, but in the case that the user asks to sort based on that field, just provide back the field we generated with the same sorted values! /bonk. So yes, this should now be working as expected :_)

**Pull request merged**